### DISCUSSION TITLE
Broken bundle depending on symlinks

### DISCUSSION THREAD

**Discussion by bug-brain** [webpack/webpack#16797] [Q&A]
During our projects migration from `yarn` to `pnpm` we encountered a problem using `monaco-editor-webpack-plugin`. It turns out that it disappeared when `resolve.symlinks` is set to `true` or C# is the only supported language.
There is [a repo](https://github.com/bug-brain/pnpm-webpack-monaco-bug) with a minimal reproduction and parts of the original and generated code. Long story short: There is something wrong with the order of dependencies; a function of a module is called before it has been fully initialized.
I expected an error during bundling in such cases. Could this be a bug in either the plugin or webpack itself?

**Comment by alexander-akait** [comment]
Hello, sorry for delay, why you have `symlinks: false` with the package manager based on symlinks...

Please remove:
```
resolve: {
  extensions: [".js"],
  symlinks: false,
},
```

You don't need `extensions: [".js"],` globally too for resolver, because you will lost resolving cjs/mjs files. I recommend do not touch this at all and override it if you need on loader level (look our documentation), expect when you really need a new extension.

You can put `console.log("test");` in `./esm/vs/editor/editor.main.js` (in `monaco-editor` package), with `symlinks: false` and without and you will see that resolving doesn't work as expected. 

if you want to debug and see why it happens you can look at loader in the plugin and will see how code generated with and without option

**Reply by bug-brain** [reply]
Both `extensions` and `symlinks` are remnants of aforementioned migration and I know they should be removed. I want to know why there is no error during the bundling process.

I added the log message and it went off when `csharp` was the only language. For me it seems the chunks are created correctly but loaded in a wrong order depending which languages are enabled.

I will try to find a difference in the generated code.

**Reply by alexander-akait** [reply]
@bug-brain The problem in resolving, run `npm ls monaco-editor`, you have different versions and webpack mixed them because can't resolve valid a version
